src/test/util/setup_common.h

```diff
@@ -39,13 +40,42 @@ extern const std::function<std::vector<const char*>()> G_TEST_COMMAND_LINE_ARGUM
 /** Retrieve the unit test name. */
 extern const std::function<std::string()> G_TEST_GET_FULL_NAME;
 
-// Enable BOOST_CHECK_EQUAL for enum class types
 namespace std {
+// Enable BOOST_CHECK_EQUAL for enum class types
 template <typename T>
-std::ostream& operator<<(typename std::enable_if<std::is_enum<T>::value, std::ostream>::type& stream, const T& e)
+inline std::ostream& operator<<(typename std::enable_if<std::is_enum<T>::value, std::ostream>::type& stream, const T& e)
 {
     return stream << static_cast<typename std::underlying_type<T>::type>(e);
 }
+
+// Enable BOOST_CHECK_EQUAL for std::optional
+template <typename T>
+inline std::ostream& operator<<(std::ostream& os, const std::optional<T>& v)
+{
+    if (v) {
+        return os << v.value();
+    } else {
+        return os << "std::nullopt";
+    }
+}
+
+// Enable BOOST_CHECK_EQUAL for arith_uint256
+inline std::ostream& operator<<(std::ostream& os, const arith_uint256& num)
+{
+    return os << num.ToString();
+}
+
+// Enable BOOST_CHECK_EQUAL for uint256
+inline std::ostream& operator<<(std::ostream& os, const uint256& num)
+{
+    return os << num.ToString();
+}
+
+// Enable BOOST_CHECK_EQUAL for uint160
+inline std::ostream& operator<<(std::ostream& os, const uint160& num)
+{
+    return os << num.ToString();
+}
```

Comment:
Why:
1. Put these 3 inside `namespace std`?
2. `inline` these non-templated functions?
   I know one day we will have modules and be happy without .H/CPP distinction, but until then, let us try to keep headers lean. Especially when the coding style mandates minimum 4 lines per function implementation, and an #include needs adding.